src/carma-streets/parse_kafka_logs.py

```diff
@@ -235,22 +235,22 @@ def parse_sdsm_to_csv(inputfile: Path, outputfile: Path, simulation: bool = Fals
     #write data of interest to csv which will be used to produce plots
     with open(outputfile, 'w', newline='') as file:
         csv_writer = writer(file)
-        csv_writer.writerow(['Created Time(ms)', 'Timestamp(ms)',
-             'Message Count', 'Source ID', 'Equipement Type', 'Reference Position Longitude', 
+        csv_writer.writerow(['system_time_ms', 'cdasim_time_ms',
```

Comment:
This may not be CDASim time. This is just the message time stamp. When specifying the `--simulation` parameter this message time stamp is interpreted without time zone and when the parameter is not specified it is interpreted with time zone.

Reply:
I think if we want a column to indicate that the message time stamps are from a simulation run that would be valid but would have to be conditional on the simulation parameter.

Reply:
Something like `Time in ms (sim)`  or `timestamp_ms(sim)` vs `timestamp_ms`. Also the created time is not the wall time translation of the timestamp, it is a timestamp of when Kafka received the message, which may be very different from message timestamp. If there was 1 second latency end to end for a vehicle produced message it would appear here as a 1 second different between `Created Time(ms)` and `Timestamp(ms)`

Reply:
lastly if we want to use snake case for column headers instead of full text I am probably indifferent to that but we should change it for all column header and for the plotting scripts too. Out of curiosity, why snake case for column headers?

Reply:
Could we have the `--simulation` flag also change the CSV header name to `cda_sim_time_ms`? Otherwise it can use `est_time_ms` or `edt_time_ms` (or `utc_time_ms` if we're using UTC time).

Keeping the time reference (e.g., `cdasim_`, `carla_`, `system_`) removes ambiguity. It also makes column names consistent with my proposed naming schemes for other columns (e.g., `map_position_x_m` or `body_twist_x_mps`).

There's no technical reason that I know of to prefer snake case (or no spaces in general). I think it looks nicer.

Reply:
I think having the flag change the csv header name is fine. The one issue I have with the header names is that they lose the information of what the time is. It is the timestamp associated with the message. This is relevant because a message with timestamp 200 may  have been sent at CDASim time 300  due to the service falling behind. Associated a CDASim time with each message is actually a process I do separately in the frequency plotter by checking wall times for when time steps were received(by kafka) and when messages were received(by kafka).  

Lastly if you want to change the column header names I am fine with that, but we should do it for all column headers in this script to be consistent and we should also fix my plotting scripts which rely on these column headers since changing them here will break those currently

Reply:
@paulbourelly999 and I had an offline discussion and decided on the following header naming changes:

* `Created Time(ms)` -> `msg_time_ms`
* `Timestamp(ms)` -> `system_time_ms`

Some justification/clarification:

* `Created Time(ms)` comes from the timestamp that travels as part of the message. The messages do not know the clock source that generates the time, so a higher-level script or user would have to imbue that information when generating CSV files from logs. The clock sources changes depending on the messages and the contexts in which CAMRA components run (real-world or simulation). Maintaining context-aware header names would be impractical, especially since most of the downstream scripts don't care about this context.
* `Timestamp(ms)` is always a system time. If CARMA runs in a simulation, these entries are the system times for the same computer. If we deploying to the real world, these values are system times for different devices. We assume that all devices have synchronized system clocks (using NTP or some other method).

Reply:
At the time of writing this comment, we have reversed the original changes. Maybe this explanation also have the definitions reversed?